iothub/device/src/InternalClient.cs

```diff
@@ -3,33 +3,34 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
+using System.Net.Sockets;
+using System.Net.WebSockets;
 using System.Threading;
 using System.Threading.Tasks;
+using DotNetty.Transport.Channels;
 using Microsoft.Azure.Devices.Client.Exceptions;
 using Microsoft.Azure.Devices.Client.Transport;
 using Microsoft.Azure.Devices.Client.Utilities;
 
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Contains methods that a device can use to send messages to and receive messages from the service,
+    /// Contains methods that a client can use to send messages to and receive messages from the service,
     /// respond to direct method invocations from the service, and send and receive twin property updates.
     /// </summary>
-    internal class InternalClient : IDisposable
+    public class InternalClient : IDisposable
     {
+        private protected readonly PipelineContext _pipelineContext;
+        private protected readonly IotHubClientOptions _clientOptions;
+
         private readonly SemaphoreSlim _methodsSemaphore = new(1, 1);
-        private readonly SemaphoreSlim _deviceReceiveMessageSemaphore = new(1, 1);
-        private readonly SemaphoreSlim _moduleReceiveMessageSemaphore = new(1, 1);
         private readonly SemaphoreSlim _twinDesiredPropertySemaphore = new(1, 1);
-        private readonly HttpTransportHandler _fileUploadHttpTransportHandler;
-        private readonly IotHubClientOptions _clientOptions;
 
         // Connection status change information
         private volatile Action<ConnectionStatusInfo> _connectionStatusChangeHandler;
 
-        internal ConnectionStatusInfo _connectionStatusInfo { get; private set; } = new ConnectionStatusInfo();
```

Comment:
Why is this disappearing from InternalClient? Wouldn't both Device and Modules have this?

Reply:
Oh, never mind. I see it further below